src/runtime/recipe/recipe-resolver.ts

```diff
@@ -89,19 +104,25 @@ export class ResolveWalker extends RecipeWalker {
       return true;
     });
 
-    if (mappable.length === 1) {
-      return (recipe, handle) => {
-        handle.mapToStorage(mappable[0]);
-        return 0;
-      };
+    if (mappable.length === 0) {
+      return error('No mappable connections (due to filtering)');
```

Comment:
I think it would be hard to understand what this error means without reading the method code, which is suboptimal.

What about something like: "The only handles matching the requested type and tags are already used in or mapped into this recipe"

Reply:
I think your message is clearer, but I am not sure that it's clear this is what we should be doing.
I wasn't attempting to change the behavior of this code though, would like to discuss it in future.

Reply:
Sure, let's discuss f2f.

There are some interesting questions that could be asked. E.g. what if you have two map statements with different tags (e.g. #homeAddress, #workAddress) and our system decided that the same handle should be mapped to both statements (e.g. because you have an office at home)? Is it ok? Is it an error? That's not obvious...